tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,500 @@
+                throw new IllegalStateException("Cannot find resources directory on the given location: " + resourcesLocation + ". Type help for explanation.");
+            }
+
+            if (!Files.isDirectory(resourcesLocation)) {
+                throw new IllegalStateException("Provided path does not represent a directory: " + resourcesLocation + ". Type help for explanation.");
+            }
+
+            return resourcesLocation;
+        });
+    }
+
+    private List<String> getDockerImages() {
+        ContributingQuestion question = questions.get("docker");
+
+        List<String> images = new ArrayList<>();
+        while (true) {
+            String nextImage = InteractiveTaskUtils.askQuestion(question.question(), question.help(), answer -> {
+                if (!answer.equalsIgnoreCase("-") && answer.split(":").length != 2) {
+                    throw new IllegalStateException("Docker image name not provided in the correct format. Type help for explanation.");
+                }
+
+                return answer;
+            });
+
+            if (nextImage.trim().equalsIgnoreCase("-")) {
+                break;
+            }
+
+            images.add(nextImage);
+        }
+
+        return images;
+    }
+
+    private List<String> getAllowedPackages() {
+        ContributingQuestion question = questions.get("allowedPackages");
+        List<String> packages = new ArrayList<>();
+        while (true) {
+            String nextPackage = InteractiveTaskUtils.askQuestion(question.question(), question.help(), answer -> answer);
+            if (nextPackage.trim().equalsIgnoreCase("-")) {
+                if (packages.isEmpty()) {
+                    InteractiveTaskUtils.printErrorMessage("At least one package must be provided. Type help for explanation.");
+                    continue;
+                }
+
+                break;
+            }
+
+            packages.add(nextPackage);
+        }
+
+        return packages;
+    }
+
+    private List<Coordinates> getAdditionalDependencies() {
+        ContributingQuestion question = questions.get("additionalDependencies");
+        List<Coordinates> dependencies = new ArrayList<>();
+        while (true) {
+            Coordinates dependency = InteractiveTaskUtils.askQuestion(question.question(), question.help(), answer -> {
+                if (answer.equalsIgnoreCase("-")) {
+                    return null;
+                }
+
+                String[] coordinatesParts = answer.split(":");
+                if (coordinatesParts.length != 3) {
+                    throw new IllegalStateException("Maven coordinates not provided in the correct format. Type help for explanation.");
+                }
+
+                String group = coordinatesParts[0];
+                String artifact = coordinatesParts[1];
+                String version = coordinatesParts[2];
+                return new Coordinates(group, artifact, version);
```

Comment:
I have seen this before. Make a `Coordinates.create(String)` method and use it where appropriate.